tests/regtest.py

```diff
@@ -1042,32 +1043,51 @@ def assertHasCaps(self, path, caps):
 			self.assertAnyCaps(path)
 			return
 
+		# deny paths containing whitespace for simplicity
+		if len(path.split()) != 1:
+			self.printError(path, "no support to check capabilities of path containing whitespace".format(
+				expected_caps, actual_caps
+			))
+
 		getcap_out = subprocess.check_output(
 			[ getcap, "-v", path ],
 			close_fds = True,
 			shell = False,
 		)
 
-		# getcap uses a '+' to indicate capability types, while
-		# permissions uses '=', so adjust accordingly
-		expected_caps = ','.join(caps).replace('=', '+')
+		expected_caps = ','.join(caps)
 		actual_caps = ""
 
+		# until libcap-2.32 the output format looked like this:
+		#
+		# /usr/bin/ping = cap_net_raw+ep
+		#
+		# starting from libcap-2.42 it looks like this:
+		#
+		# /usr/bin/cdrecord cap_ipc_lock,cap_sys_rawio,cap_sys_nice,cap_sys_resource=ep
+		#
+		# see bsc#1175076 comment 2.
+		# So let's be agnostic to the output format.
+
 		# output is something like "/path/to/file = cap_stuff+letters"
 		for line in getcap_out.decode('utf8').splitlines():
-			# be prudent about possible spaces or equals in paths,
-			# even though it should never occur in our test
-			# environment
-			parts = line.split('=')
-			if len(parts) < 2:
+			parts = line.split()
+			if parts[0] != path:
+				continue
+			elif len(parts) < 2:
 				continue
 
-			cap_path = '='.join(parts[:-1]).strip()
-			if cap_path != path:
-				# not for our file
+			if len(parts) == 3 and parts[1] == '=':
+				# the old output format
+				# getcap uses a '+' to indicate capability
+				# types, while permissions uses '=', so adjust
+				# accordingly
+				expected_caps = expected_caps.replace('=', '+')
+				parts.pop(1)
```

Comment:
Does the `pop` do anything here?

Reply:
Ah, now I see - you're removing the thing at index 1. Could you instead move the assignment of `parts[1]` into the different branches, please?